src/mame/jaleco/jaleco_vj_pc.cpp

```diff
@@ -0,0 +1,167 @@
+{
+    device.option_add("fdc37c93x", FDC37C93X);
+}
+
+static void isa_com(device_slot_interface &device)
+{
+    // TODO: Stepping Stage uses a different UPS but will still boot unlike VJ without a valid UPS attached
+    device.option_add("jaleco_vj_ups", JALECO_VJ_UPS);
+}
+
+static void isa_cards(device_slot_interface &device)
+{
+    device.option_add("jaleco_vj_sound", JALECO_VJ_ISA16_SOUND);
+}
+
+void jaleco_vj_pc_device::superio_config(device_t *device)
+{
+    // TODO: This should be a Winbond W83877F Super I/O
+    fdc37c93x_device &fdc = *downcast<fdc37c93x_device *>(device);
+    fdc.set_sysopt_pin(1);
+    fdc.gp20_reset().set_inputline(m_maincpu, INPUT_LINE_RESET);
+    fdc.gp25_gatea20().set_inputline(m_maincpu, INPUT_LINE_A20);
+    fdc.irq1().set(":jaleco_vj_pc:pci:07.0", FUNC(i82371sb_isa_device::pc_irq1_w));
+    fdc.irq8().set(":jaleco_vj_pc:pci:07.0", FUNC(i82371sb_isa_device::pc_irq8n_w));
+    fdc.txd1().set(":jaleco_vj_pc:serport0", FUNC(rs232_port_device::write_txd));
+    fdc.ndtr1().set(":jaleco_vj_pc:serport0", FUNC(rs232_port_device::write_dtr));
+    fdc.nrts1().set(":jaleco_vj_pc:serport0", FUNC(rs232_port_device::write_rts));
+    fdc.txd2().set(":jaleco_vj_pc:serport1", FUNC(rs232_port_device::write_txd));
+    fdc.ndtr2().set(":jaleco_vj_pc:serport1", FUNC(rs232_port_device::write_dtr));
+    fdc.nrts2().set(":jaleco_vj_pc:serport1", FUNC(rs232_port_device::write_rts));
+}
+
+void jaleco_vj_pc_device::boot_state_w(uint8_t data)
+{
+    logerror("Boot state %02x\n", data);
+}
+
+void jaleco_vj_pc_device::device_add_mconfig(machine_config &config)
+{
+    PENTIUM(config, m_maincpu, 20'000'000); // TODO: Pentium 60/90mhz? Underclocked for performance reasons
+    m_maincpu->set_irq_acknowledge_callback("pci:07.0:pic8259_master", FUNC(pic8259_device::inta_cb));
+    m_maincpu->smiact().set("pci:00.0", FUNC(i82439hx_host_device::smi_act_w));
+
+    PCI_ROOT(config, "pci", 0);
+    I82439HX(config, "pci:00.0", 0, m_maincpu, 256*1024*1024); // TODO: Should be 0x05851106 VIA VT82C585 Apollo VP,VPX,VPX-97 System Controller
+
+    i82371sb_isa_device &isa(I82371SB_ISA(config, "pci:07.0", 0, m_maincpu));
+    // isa.set_ids(0x05861106, 0x23, 0x060100, 0x00000000); // TODO: Should be VIA VT82C586B, PCI-to-ISA Bridge
+    isa.boot_state_hook().set(FUNC(jaleco_vj_pc_device::boot_state_w));
+    isa.smi().set_inputline(m_maincpu, INPUT_LINE_SMI);
+
+    i82371sb_ide_device &ide(I82371SB_IDE(config, "pci:07.1", 0, m_maincpu));
+    // ide.set_ids(0x05711106, 0x06, 0x01018a, 0x00000000); // TODO: Should be VIA VT82C586B, IDE Controller
+    ide.irq_pri().set("pci:07.0", FUNC(i82371sb_isa_device::pc_irq14_w));
+    ide.irq_sec().set("pci:07.0", FUNC(i82371sb_isa_device::pc_mirq0_w));
+
+    // TODO: pci:07.3 0x30401106 VIA VT83C572, VT86C586/A/B Power Management Controller
+
+    VIRGEDX_PCI(config, "pci:10.0", 0);
+
+    jaleco_vj_king_qtaro_device& king_qtaro(JALECO_VJ_KING_QTARO(config, "pci:08.0", 0));
```

Comment:
Out of curiosity: are these PCI config IDs trusted somehow or it's just works enough, next?

Reply:
I pulled all of the PCI config IDs and device PID/VIDs and such out of the Windows registery, so these all should be accurate to the actual hardware to the best of my understanding.

Reply:
For reference, `HKEY_LOCAL_MACHINE\Enum\PCI\` is where I was looking:
![image](https://user-images.githubusercontent.com/63495610/224336678-f9309bb2-f7a1-47e1-983d-8c3ee8a5f4a7.png)

Reply:
S3 Virge DX is wrong then, it should be "pci:0a.0" not "pci:10.0". Please also add a note on top of the machine config about the routing being trusted by the HDD regedit output.

Reply:
I tried various IDs but "pci:0a.0" causes the video output to stop working entirely. After some testing, everything in the 09 to 0e range resulted in a black screen but 0f and 10 and 11 worked. I have no idea what the cause of that is so I just added the proper ID as another TODO.